app/controllers/concerns/reports_actions.rb

```diff
@@ -62,4 +62,35 @@ def file_timestamp
   def i18n_scope
     'admin.reports'
   end
+
+  def rendering_options
+    @rendering_options ||= ReportRenderingOptions.where(
+      user: spree_current_user,
+      report_type: report_type,
+      report_subtype: report_subtype
+    ).first_or_create do |report_rendering_options|
+      report_rendering_options.options[:fields_to_show] = if request.get?
+                                                            @report.columns.keys -
+                                                              @report.fields_to_hide
+                                                          else
+                                                            params[:fields_to_show]
+                                                          end
+      report_rendering_options.options[:display_summary_row] = request.get?
+      report_rendering_options.options[:display_header_row] = false
```

Comment:
Looking over this again, I can see that this section could also be DRY'ed up like below, eg:

```
report_rendering_options.options = {
    fields_to_show: if request.get? #...etc
    display_summary_row: #..etc
}
```

Reply:
@dacook  done :)